internals/overlord/state/notices.go

```diff
@@ -229,6 +273,9 @@ func (s *State) AddNotice(noticeType NoticeType, key string, options *AddNoticeO
 		newOrRepeated = true
 	} else {
 		// Additional occurrence, update existing notice
+		if !userIDsEqual(options.UserID, notice.userID) {
+			return "", fmt.Errorf("existing notice user ID %s does not match that of new notice with the same type and key: %s", optionIntToString(notice.userID), optionIntToString(options.UserID))
```

Comment:
This error message is a bit of mouthful. How about this:

```suggestion
			return "", fmt.Errorf("user ID %s does not match existing notice's (%s)", optionIntToString(options.UserID), optionIntToString(notice.userID))
```

Reply:
Much cleaner, thanks!